.github/workflows/manual-deploy-staging.yaml

```diff
@@ -0,0 +1,80 @@
+name: Manual Deploy to Staging
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: tag
+        required: true
+
+env:
+  environment: staging
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: staging
+    steps:
+    - name: Clone latest repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Check that the tag exists in repo
+      id: tag_check
+      run: |
+        if git rev-parse 'refs/tags/${{ github.event.inputs.tag }}' &> /dev/null; then
+          echo '::set-output name=tag::${{ github.event.inputs.tag }}'
+        elif git rev-parse '${{ github.event.inputs.tag }}' &> /dev/null; then
+          echo "::set-output name=tag::`git rev-parse '${{ github.event.inputs.tag }}'`"
+        else
+          echo "Couldn't figure out tag from input: ${{ github.event.inputs.tag }}"
+          echo "Aborting deployment."
+          false
+        fi
+    - name: Log into Github Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Check that the tag exists in container registry
+      id: image_check
+      run: |
+        if docker manifest inspect ghcr.io/hathitrust/otis-unstable:${{ steps.tag_check.outputs.tag }} > /dev/null; then
+          echo '::set-output name=image_exists::true'
+          echo "image exists!"
+        else
+          echo "image doesn't exist; Starting to Build and push image"
+        fi
+    - name: Checkout Correct repository
+      if: ${{ steps.image_check.outputs.image_exists != 'true' }}
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.inputs.tag }}
+    - name: Set up QEMU
+      if: ${{ steps.image_check.outputs.image_exists != 'true' }}
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      if: ${{ steps.image_check.outputs.image_exists != 'true' }}
+      uses: docker/setup-buildx-action@v1
+    - name: Build and Push
+      if: ${{ steps.image_check.outputs.image_exists != 'true' }}
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        file: Dockerfile.prod
+        # This (and the QEMU/buildx setup) may not be needed if we only want to get
```

Comment:
Right, it's not an issue in k8s; it only comes up when we want to reuse these images for development on the desktop. I think my inclination is just to include multiarch by default to avoid the potential for confusion down the line. If eventually we can avoid duplication across actions/workflows the only reason to avoid it is if it's particularly time-consuming to do so & we're trying to cut down on our github actions CPU usage. I don't think we're particularly close to hitting that limit any time soon.

Reply:
Agreed. That said, be aware multiarch builds take significantly longer than the single-arch "native" builds we've been doing. 18-22 minutes vs the previous 3-ish.

Reply:
Hmm, OK. I was hoping for just double instead of 7x, but I guess that makes sense if it's via emulation for arm. We can give it a try and see what we think, I guess.

Reply:
C compiler or compiler-adjacent (like `ldconfig`)? I dunno. But it sure smells like QEMU is the time hog. With our `n=2` samples.